File: java/src/main/java/it/giannibombelli/wsc2026/giftcard/application/commands/IssueGiftCard.java
package it.giannibombelli.wsc2026.giftcard.application.commands;

import it.giannibombelli.wsc2026.common.utils.Require;

import it.giannibombelli.wsc2026.common.application.Command;
import it.giannibombelli.wsc2026.giftcard.domain.giftcard.GiftCardId;

public record IssueGiftCard(GiftCardId aggregateId) implements Command<GiftCardId> {
    public IssueGiftCard {
        Require.requireArgument(aggregateId, "giftCardId");
    }
}

File: typescript/src/giftcard/application/commands/issueGiftCard.ts
// Command per emettere una nuova gift card.

import type { Command } from '@/common/application/command.js';
import type { GiftCardId } from '../../domain/giftcard/giftCardId.js';
import { requireArgument } from '@/common/utils/requireArgument.js';

export type IssueGiftCard = Command<'IssueGiftCard', GiftCardId>;

export function issueGiftCard(aggregateId: GiftCardId): IssueGiftCard {
  requireArgument(aggregateId, 'giftCardId');
  return { kind: 'IssueGiftCard', aggregateId };
}
